src/track/serato/beatgrid.cpp

```diff
@@ -417,6 +417,97 @@ QByteArray SeratoBeatGrid::dumpBase64Encoded() const {
     return base64EncodedData;
 }
 
+void SeratoBeatGrid::setBeats(BeatsPointer pBeats,
+        const audio::StreamInfo& streamInfo,
+        double timingOffsetMillis) {
+    VERIFY_OR_DEBUG_ASSERT(pBeats) {
+        return;
+    }
+
+    const double timingOffsetSecs = timingOffsetMillis / 1000;
+    SINT endPositionSamples = streamInfo.getSignalInfo().frames2samples(
```

Comment:
This position might behind behind the actual available audio data. I think it is worth noting here.